pgml-extension/pgml_extension/model.py

```diff
@@ -729,15 +816,43 @@ def fit(self, snapshot: Snapshot):
                     roc_auc_y_prob = y_prob[:, 1]
                 metrics["roc_auc"] = roc_auc_score(y_test, roc_auc_y_prob, average="weighted", multi_class="ovo")
 
-        # Save the model
+        self.metrics = metrics        
+
+        # Save the results
+        if not os.path.exists(self.path):
+            os.makedirs(self.path)
+        with open(self.pickle_path, "wb") as f:
```

Comment:
Not ACID, can silently break because of lack of fsync call, cannot be replicated to replicas.